### ISSUE TITLE
Comments: SSO login not working in Chrome

### ISSUE THREAD

**Issue by KokkieH** [Automattic/jetpack#15030]
#### Steps to reproduce the issue

1. In Chrome, visit a post with comments enabled and click on the comments field.
2. If you're shown as logged into WordPress.com, click Log out or Change, and try logging in via Twitter, Facebook, or WordPress.com

#### What I expected

To show as logged into the comment form using the social media account, and able to comment.

#### What happened instead
The Connecting indicator appears, but the connection never completes.

The login only completes using the Google login option.

This only appears to happen in Chrome - I can login using those services without problem in Firefox and Safari. Tested on Chrome 80.0.3987.132 (Official Build) (64-bit) on Mac OS 10.14.6

I don't see any Console errors

I was able to replicate the issue on my own Jetpack site, but not on a WordPress.com simple site.

**Screenshots**
![comments](https://user-images.githubusercontent.com/11873759/76946836-546ad400-690d-11ea-8c51-18f5ea47c234.gif)

Reported in the forums at https://wordpress.org/support/topic/error-comentarios-jetpack/

**Comment by jeherve** [comment]
Are you able to reproduce the issue on WordPress.com sites as well?

**Comment by KokkieH** [comment]
@jeherve no, only on a Jetpack site. On a WordPress.com simple site I can connect without problems in Chrome.

**Comment by jeherve** [comment]
Do you happen to have a WordPress.com site with a mapped domain where you could test this? I'd like to confirm that this is indeed an issue with the changes to domain cookies added by Chrome last month.

**Comment by KokkieH** [comment]
I'd tested it on kokkieh.blog, so it does work with a mapped domain, yes. Also on a site with a free WordPress.com URL.

**Comment by jeherve** [comment]
Thank you! Can you reproduce here?
https://jeremy.hu/our-small-animals/

I can't seem to be able to reproduce at the moment.

**Comment by KokkieH** [comment]
I can, yes. Checked both in a regular Chrome session and incognito - the connecting indicator just keeps running.

Chrome has since updated to 80.0.3987.149 as well. No difference after a relaunch.

**Comment by eduvalruiz** [comment]
I have the same proble. wouldn´t it be better to report a Jetpack?

**Comment by michalpelc1** [comment]
I have the same with the following messages:
- FB "Connecting to Facebook"
- Google "Error: Google failed to return an expected code."

System: Windows 10, build 1909 18363.720
Chrome: 80.0.3987.149
I have also tested with Firefox and I didn't get these error messages.

**Comment by bindlegirl** [comment]
We seem to have the same problem in Safari. @jeherve should I open a new issue for this or adding it here OK.

Safari version is 13.1

**Comment by jeherve** [comment]
> should I open a new issue for this or adding it here OK.

Here is fine 👍

**Comment by AsifMahsud** [comment]
Same probelm with jetpack comments... 
It gives the error message "Error: Google failed to return an expected code." when trying to comment using Google account.
Link: https://techinspec.com/artificial-intelligence-has-learned-how-to-make-perfect-sandwiches/
I'm trying to figure out how to report this issue to JetPack officials.

**Comment by nithinmanne** [comment]
Is there any progress on this? I'm facing the same issue as well.
Link: https://mutedmuse.com

**Comment by hitolonen** [comment]
Experiencing this as well on a non-WP.com.

Firefox - no issues with more or less basic settings.

Safari - fix was to disable Prevent cross-site tracking - Safari --> Preferences --> Privacy, disable the "Prevent cross-site tracking" option.

Chrome - can replicate in both incognito and non-incognito. Fiddling cookie/privacy settings does nothing. The loading icon simply stays on, reloading the page shows the login was successful.  Console > Issues shows notes about cookie's SameSite attribute and suggests `SameSite=None` fix.

**Comment by hitolonen** [comment]
-

**Comment by jeherve** [comment]
> Console > Issues shows notes about cookie's SameSite attribute and suggests SameSite=None fix.

Could you tell me more about this? What was the domain that was impacted by the warning, and at what point did you see it popping up?

**Comment by hitolonen** [comment]
@jeherve 

> Could you tell me more about this?

On page load Console > Issues shows 17/18 notices, 12/13 cookies, 11 requests.

Once the login process is happening, 5 new issues popup.

<img width="1612" alt="chrome console - issues" src="https://user-images.githubusercontent.com/20572314/92398405-284c4500-f131-11ea-95c9-b56c3261e9d4.png">

**Comment by hitolonen** [comment]
Chrome workaround: Set `SameSite by default cookies` to `disabled`. This disables Chrome's default behavior of setting `SameSite=Lax`.

**Comment by jeherve** [comment]
This should now be fixed as of D49592-code. 🎉 

If this keeps happening, let us know!

**Comment by MicahBurke** [comment]
Still happening. Version 9.2.1

**Comment by htdat** [comment]
@jeherve - it seems that this issue happens again. 

I can replicate it with Jetpack 9.2.1 and the latest Chrome `Version 87.0.4280.88 (Official Build) (x86_64)`: 

Twitter https://d.pr/i/l2QZj0
Facebook https://d.pr/i/49xVAx
Google https://d.pr/i/UgSNEq

**Comment by leonbloy** [comment]
Same happening in this one: https://markpshea.com/2021/01/09/my-friend-sherry-weddell-on-intentional-and-fruitful-discipleship/#comments

**Comment by danjjohnson** [comment]
Also reported here: https://wordpress.org/support/topic/infinite-load-facebook-social-login-jetpack-comment/

**Comment by jeherve** [comment]
Also in 3668688-zen

**Comment by anaemnesis** [comment]
Reported in 122692-zen (VIP)

**Comment by archzaiko** [comment]
Still happening. Version 9.4

**Comment by danjjohnson** [comment]
Also reported here: https://wordpress.org/support/topic/social-login-issue-2/

**Comment by danjjohnson** [comment]
3773089-zen

**Comment by magdigit** [comment]
This error is still there with my website and it gives me a prompt "Error: Google failed to return an expected code." Do we have a solution yet, or I shall disable the jetpack comment, I love this plugin but the comment error is hurting a lot. This is my website [https://magdigit.com/](url) and I tested with the latest browser version of Chrome, Mozilla, Edge and Opera.

**Comment by danjjohnson** [comment]
3881522-zen

**Comment by lakellie** [comment]
Another report in 3966115-zen

**Comment by lakellie** [comment]
Another report in 3982475-zen, getting the `Error: Google failed to return an expected code` error

**Comment by kokoglen** [comment]
Im having this problem as well. https://commadot.com
It started happening after I upgraded the server to PHP 8.
What is the status? This is a big problem.

**Comment by jeherve** [comment]
> What is the status? This is a big problem.

I'm afraid we do not have an update yet. Until this is fixed, and if you experience the issue on your site, I would recommend that you disable Jetpack's Comment form feature under Jetpack > Settings > Discussion in your dashboard.

We'll update this issue as soon as this gets fixed.

**Comment by kokoglen** [comment]
It's been a month. Can you at least describe the problem? Is there a different comments plugin that people suggest?

**Comment by jeherve** [comment]
> Can you at least describe the problem?

We do not have many details at the moment, since the problem cannot easily be reproduced. As an example, I appear to be able to use the different options without any issues on your site, using Chrome Version 91.0.4472.114 (Official Build) (arm64):

<img width="875" alt="Screenshot 2021-06-28 at 10 04 59" src="https://user-images.githubusercontent.com/426388/123602177-c6141c00-d7f8-11eb-8f0c-2e51f63dd798.png">
<img width="1164" alt="Screenshot 2021-06-28 at 10 04 48" src="https://user-images.githubusercontent.com/426388/123602198-cad8d000-d7f8-11eb-92cf-199ff19fd555.png">

> Is there a different comments plugin that people suggest?

I think this mostly depends on what social login options you're most interested in on your site.

**Comment by danjjohnson** [comment]
Another report here: https://wordpress.org/support/topic/comments-by-social-media-accounts-not-working/

**Comment by danjjohnson** [comment]
Another here: https://wordpress.org/support/topic/connecting-to/

**Comment by jeherve** [comment]
This no longer seems to be an issue. Closing.
